src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -1014,13 +1009,12 @@ static PyObject* nrnpy_set_psection_safe(PyObject* self, PyObject* args) {
 
 static PyObject* NPySecObj_psection(NPySecObj* self) {
     CHECK_SEC_INVALID(self->sec_);
-    if (nrnpy_psection) {
-        PyObject* arglist = Py_BuildValue("(O)", self);
-        PyObject* result = PyObject_CallObject(nrnpy_psection, arglist);
-        Py_DECREF(arglist);
-        return result;
+    if (!nrnpy_psection) {
+        Py_RETURN_NONE;
     }
-    Py_RETURN_NONE;
+    // Build temp tuple and track the new ref
+    auto arglist = nb::steal(Py_BuildValue("(O)", self));
+    return PyObject_CallObject(nrnpy_psection, arglist.ptr());
```

Comment:
Yep. I left this one since it's very small. In the other occurrences I tried to bring changes to a minimal set. I still kept some fixes (like the dict building which fixes the problem with PyNone not ever being IncRef'd) so please bear with me. I hope this change is still pretty straightforward to review